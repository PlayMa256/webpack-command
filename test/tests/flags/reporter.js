const strip = require('strip-ansi');

const { apply, build, crcDist, test, validate } = require('../../util');

// TODO: test stylish reporter
test('--reporter', module, () => {
  for (const name of ['basic', 'stylish']) {
    const fixture = `reporter/${name}`;
    const opts = { fixture };

    let config;

    it(`${name} reporter should validate`, () => {
      expect(validate(opts)).toEqual(true);
    });

    it(`${name} reporter should apply`, () => {
      config = apply(opts);

      expect(config).toMatchSnapshot();
    });

    it(`${name} reporter should build`, () =>
      build(config).then((result) => {
        expect(
          strip(result).replace(/Î”t \d+ms/, '<duration>')
        ).toMatchSnapshot();
        expect(crcDist()).toMatchSnapshot();
      }));
  }
});
