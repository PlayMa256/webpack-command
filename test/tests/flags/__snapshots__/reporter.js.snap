// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --reporter > basic reporter should apply #0 1`] = `
Object {
  "context": "<PROJECT_ROOT>",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "basic",
  "stats": Object {
    "builtAt": false,
    "hash": false,
    "timings": false,
  },
}
`;

exports[`Flags > --reporter > basic reporter should build #0 1`] = `
"Version: webpack 4.3.0
  Asset      Size  Chunks             Chunk Names
main.js  2.96 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --reporter > basic reporter should build #1 1`] = `"bf58edd8"`;

exports[`Flags > --reporter > stylish reporter should apply #0 1`] = `
Object {
  "context": "<PROJECT_ROOT>",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
}
`;

exports[`Flags > --reporter > stylish reporter should build #0 1`] = `
"
webpack v4.3.0

2c2cd86350a8b0063c7b
  size     name        module                             status
  41 B     entry-a.js  ./test/fixtures/common/entry-a.js  built

  size     name        asset                              status
  3.03 kB  main        main.js                            emitted

  <duration> 
"
`;

exports[`Flags > --reporter > stylish reporter should build #1 1`] = `"bf58edd8"`;
